src/stdlib_stats_moment_all.fypp

```diff
@@ -31,10 +32,11 @@ contains
         n = real(size(x, kind = int64), ${k1}$)
 
         if (present(center)) then
-         res = sum((x - center)**order) / n
+          center_ = center
         else
-         res = sum((x - mean(x))**order) / n
+          center_ = mean(x)
         end if
+        res = sum((x - center_)**order) / n
```

Comment:
Why not (probably due to inefficiency?):
```fortran
res = sum(( x - optval(center, mean(x))**order) / n
```

Reply:
I thought the same, and I worried that maybe some compiler may place an `optval()` call in every iteration of the loop, that wouldn't be good. But I don't know. Evaluating the scalar `center_` first seems safer.

Reply:
Sorry, I read it again and now understand what you mean. Ignore my previous comment. I don't think my question/suggestion was good anymore, because `mean(x)` would be evaluated if refactored to `optval()`, so let's not do it.

Reply:
Indeed. I tested it with a large array and gfortran was slower every time, even when enabling `-Ofast -flto`. `optval` doesn't seem to be efficient here.

Reply:
thank yoju for testing. Futhermore, using `center_` __might__ add an additional temporary array, especially for the case when `center` is present. Therefore, for the different modules of `stdlib_stats_moment`, I suggest to revert the changes and to keep the original version.

Reply:
I don't understand `fypp`... is there a way for `center` to be an array? If it is only scalar the compiler should be able to optimise it.

Reply:
In this case, `center` is indeed a scalar. In `stdlib_stats_moment.fypp`, there is a similar case, however `center` is an array.

I agree that, since it is a scalar in this case, the potential cost is not high, and the compiler should be able to optimise it.